packages/lib/src/IdAllocator.ts

```diff
@@ -0,0 +1,42 @@
+export default class IdAllocator {
+  private static registry: Array<IdAllocator> = [];
+
+  private value = 0;
+
+  constructor(private prefix: string) {
+    IdAllocator.registry.push(this);
+  }
+
+  generate(): string {
+    return `${this.prefix}-${this.value++}`;
+  }
+
+  /* The following are only for testing!! */
```

Comment:
Wonder if we prefix the name to be clear: `testing_snapshot`? Maybe over kill...

Reply:
Also where are the tests that are using these?

Reply:
I was relying on the SSR tests warnings to surface these, but I'll add some unit tests

Reply:
Mind adding some JSDocs to explain what the below methods do?